presto-geospatial/src/test/java/com/facebook/presto/plugin/geospatial/TestSphericalGeoFunctions.java

```diff
@@ -201,6 +201,59 @@ public void testArea()
 
     private void assertArea(String wkt, double expectedArea)
     {
-        assertFunction(format("ABS(ROUND((ST_Area(to_spherical_geography(ST_GeometryFromText('%s'))) / %f - 1 ) * %d, 0))", wkt, expectedArea, 10000), DOUBLE, 0.0);
+        assertFunctionImprecise(format("ST_Area(to_spherical_geography(ST_GeometryFromText('%s')))", wkt), expectedArea, 1e-4);
     }
+
+    @Test
+    public void testLength()
+    {
+        double x1 = -71.05, y1 = 42.36,   // Boston
+               x2 = -87.62, y2 = 41.87,   // Chicago
+               x3 = -122.41, y3 = 37.77;  // San Francisco
+
+        double distance1 = 1365691.096620299, distance2 = 2985175.539584118;
+        double length = distance1 + distance2;
+
+        // Empty linestring returns null
+        assertLength("LINESTRING EMPTY", null);
+
+        // Linestring with one point has length 0
+        assertLength(format("LINESTRING (%.2f %.2f)", x1, y1),0.0);
+
+        // Linestring with only one distinct point has length 0
+        assertLength(format("LINESTRING (%.2f %.2f, %.2f %.2f, %.2f %.2f)", x1, y1, x1, y1, x1, y1), 0.0);
+
+        // ST_Length is equivalent to sums of ST_DISTANCE between points in the LineString
+        assertDistance(format("POINT (%.2f %.2f)", x1, y1), format("POINT (%.2f %.2f)", x2, y2), distance1);
+        assertDistance(format("POINT (%.2f %.2f)", x2, y2), format("POINT (%.2f %.2f)", x3, y3), distance2);
+        assertLength(format("LINESTRING (%.2f %.2f, %.2f %.2f, %.2f %.2f)", x1, y1, x2, y2, x3, y3), length);
```

Comment:
I find this difficult to read. Hardcoding coordinates and expected length will be clearer.